src/main/java/com/ministore/api/controllers/GetUserHandler.java

```diff
@@ -23,7 +26,13 @@ public GetUserHandler(UserRepository repository) {
         produces = "application/json"
     )
     public Optional<User> getUser(@PathVariable String id) {
-        return this.repository.findById(id);
+        Optional<User> user = this.repository.findById(id);
+
+        if (user.isEmpty()) { 
```

Comment:
A good alternative would be to remove this `if` check and use the [`user.orElseThrow()`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-) method instead.

Let me know if you have any questions on how to do it.